src/main/java/io/fabric8/elasticsearch/plugin/filter/FieldStatsResponseFilter.java

```diff
@@ -72,6 +73,17 @@ public void apply(final Task task, final String action, final ActionRequest requ
             @SuppressWarnings("unchecked")
             @Override
             public void onResponse(ActionResponse response) {
+                if(INDICES_FIELD_STATS_READ_ACTION.equals(action) && response instanceof FieldStatsResponse) {
+                    if(((FieldStatsResponse)response).getIndicesMergedFieldStats().size() == 0) {
+                        LOGGER.trace("Modifying the response to be {}", RestStatus.NOT_FOUND);
```

Comment:
Should this trace message be ' RestStatus.NO_CONTENT'

Reply:
yes I will fix it.